tests/test_subdomains.py

```diff
@@ -622,3 +622,32 @@ class Dummy(SubDomainSet):
 
         assert_structure(op, ['t,n', 't,n,xi_n0_blk0,yi_n0_blk0,xi_n,yi_n,zi_n'],
                          't,n,xi_n0_blk0,yi_n0_blk0,xi_n,yi_n,zi_n')
+
+    def test_sequential_implicit(self):
+        """
+        Make sure implicit dimensions define sequential iteration spaces. This is for
+        performance and potentially for correctness too (e.g., canonical openmp loops
+        forbid subiterators, which could potentially be required by a MultiSubDomain).
+        """
+
+        class Dummy(SubDomainSet):
+            name = 'dummy'
+
+        dummy = Dummy(N=0, bounds=[(), (), (), (), (), ()])
+
+        grid = Grid(shape=(10, 10, 10), subdomains=(dummy,))
+
+        f = TimeFunction(name='f', grid=grid, save=10)
+
+        eqn = Eq(f, 1., subdomain=grid.subdomains['dummy'])
+
+        op = Operator(eqn)
+
+        iterations = FindNodes(Iteration).visit(op)
+        assert len(iterations) == 5
```

Comment:
assert length is redundant since the assignment takes place in the next line?
You can also use assert_structure